uber/models.py

```diff
@@ -1791,6 +1799,60 @@ def html(self):
             return SafeString(self.body.replace('\n', '<br/>'))
 
 
+class PromoCode(MagModel):
+    expiration_date = Column(UTCDateTime, default=c.ESCHATON)
+    price = Column(Integer, default=0)
+    code = Column(UnicodeText)
+    old_code = ""
+    # Uses should be -1 to make expired
+    uses = Column(Integer, nullable=True, default=None)
+    used_by = relationship(Attendee)
+
+    def generate_code(self, count):
+        code = []
+        for x in range(count):
+            code.append(c.PROMO_CODE_WORDS[random.choice(string.ascii_letters + string.digits)])
+        return "_".join(code)
```

Comment:
Sorry, to clarify, you seem to be joining the promo code with underscores just to unjoin it later. Was there a reason do that? Was having it as an array throwing errors?